src/psyclone/psyir/transformations/omp_loop_trans.py

```diff
@@ -0,0 +1,261 @@
+    ...     subroutine my_subroutine()
+    ...         integer, dimension(10, 10) :: A
+    ...         integer :: i
+    ...         integer :: j
+    ...         do i = 1, 10
+    ...             do j = 1, 10
+    ...                 A(i, j) = 0
+    ...             end do
+    ...         end do
+    ...     end subroutine
+    ...     """)
+    >>> loop = psyir.walk(Loop)[0]
+    >>> omplooptrans1 = OMPLoopTrans(omp_schedule="dynamic",
+    ...                              omp_directive="paralleldo")
+    >>> omplooptrans1.apply(loop)
+    >>> print(FortranWriter()(psyir))
+    subroutine my_subroutine()
+      integer, dimension(10,10) :: a
+      integer :: i
+      integer :: j
+      integer :: th_idx
+      integer :: nthreads
+    <BLANKLINE>
+      !$omp parallel do default(shared), private(i,j), schedule(dynamic)
+      do i = 1, 10, 1
+        do j = 1, 10, 1
+          a(i,j) = 0
+        enddo
+      enddo
+      !$omp end parallel do
+    <BLANKLINE>
+    end subroutine my_subroutine
+    <BLANKLINE>
+
+    '''
+    def __init__(self, omp_directive="do", omp_schedule="auto"):
+        super().__init__()
+        # Whether or not to generate code for (run-to-run on n threads)
+        # reproducible OpenMP reductions. This setting can be overridden
+        # via the `reprod` argument to the apply() method.
+        self._reprod = Config.get().reproducible_reductions
+
+        # Use setters to set up attributes
+        self._omp_schedule = ""
+        self.omp_schedule = omp_schedule
+
+        self._omp_directive = ""
+        self.omp_directive = omp_directive
+
+    def __str__(self):
+        return "Adds an OpenMP directive to parallelise the target loop"
+
+    @property
+    def omp_directive(self):
+        '''
+        :returns: the value of the omp_directive attribute.
+        :rtype: str
+        '''
+        return self._omp_directive
+
+    @omp_directive.setter
+    def omp_directive(self, value):
+        '''
+        :param str value: new value of the omp_directive attribute.
+
+        :raises TypeError: if the provided value is not a valid str.
+        '''
+        if not isinstance(value, str) or value not in VALID_OMP_DIRECTIVES:
+            raise TypeError(
+                f"The {type(self).__name__}.omp_directive property must be "
+                f"a str with the value of {VALID_OMP_DIRECTIVES}"
+                f" but found a '{type(value).__name__}' with value '{value}'.")
+        self._omp_directive = value
+
+    @property
+    def omp_schedule(self):
+        '''
+        :returns: the OpenMP schedule that will be specified by \
+            this transformation. The default schedule is 'static'.
+        :rtype: str
+
+        '''
+        return self._omp_schedule
+
+    @omp_schedule.setter
+    def omp_schedule(self, value):
+        '''
+        :param str value: Sets the OpenMP schedule value that will be \
+            specified by this transformation.
```

Comment:
"...unless adding an OMP Loop directive (in which case it is not applicable)."?

Reply:
Done